.travis.yml

```diff
@@ -1,174 +1,261 @@
-      dart: dev
-    - stage: unit_test
-      script: ./tool/travis.sh command_5
+      cache:
+        directories:
+          - angular_forms/.dart_tool
+
+    - stage: testing
+      script: ./tool/travis.sh test
       env: PKG="angular_forms"
-      dart: dev
-    - stage: analyze
-      script: ./tool/travis.sh dartanalyzer_2
-      env: PKG="angular_router"
-      dart: dev
-    - stage: build
-      script: ./tool/travis.sh command_0
+      cache:
+        directories:
+          - angular_forms/.dart_tool
+      addons:
+        chrome: stable
+      before_install:
+        - export DISPLAY=:99.0
+        - sh -e /etc/init.d/xvfb start
+        - "t=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done"
+
+    - stage: presubmit
+      script: ./tool/travis.sh analyze
       env: PKG="angular_router"
-      dart: dev
-    - stage: unit_test
-      script: ./tool/travis.sh test_1
+
+    - stage: building
+      script: ./tool/travis.sh build
       env: PKG="angular_router"
-      dart: dev
-    - stage: unit_test
-      script: ./tool/travis.sh command_5
+      cache:
+        directories:
+          - angular_router/.dart_tool
+
+    - stage: testing
+      script: ./tool/travis.sh test
       env: PKG="angular_router"
-      dart: dev
-    - stage: analyze
-      script: ./tool/travis.sh dartanalyzer_2
-      env: PKG="angular_test"
-      dart: dev
-    - stage: build
-      script: ./tool/travis.sh command_0
+      cache:
+        directories:
+          - angular_router/.dart_tool
+      addons:
+        chrome: stable
+      before_install:
+        - export DISPLAY=:99.0
+        - sh -e /etc/init.d/xvfb start
+        - "t=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done"
+
+    - stage: presubmit
+      script: ./tool/travis.sh analyze
       env: PKG="angular_test"
-      dart: dev
-    - stage: unit_test
-      script: ./tool/travis.sh test_1
+
+    - stage: building
+      script: ./tool/travis.sh build
       env: PKG="angular_test"
-      dart: dev
-    - stage: unit_test
-      script: ./tool/travis.sh command_6
+      cache:
+        directories:
+          - angular_test/.dart_tool
+
+    - stage: testing
+      script: ./tool/travis.sh test
       env: PKG="angular_test"
-      dart: dev
-    - stage: analyze
-      script: ./tool/travis.sh dartanalyzer_4
+      cache:
+        directories:
+          - angular_test/.dart_tool
+      addons:
+        chrome: stable
+      before_install:
+        - export DISPLAY=:99.0
+        - sh -e /etc/init.d/xvfb start
+        - "t=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done"
+
+    - stage: presubmit
+      script: ./tool/travis.sh analyze
       env: PKG="examples/hacker_news_pwa"
-      dart: dev
-    - stage: build
-      script: ./tool/travis.sh command_7
+
+    - stage: building
+      script: ./tool/travis.sh build
```

Comment:
should be `build:release`?

Reply:
@matanlurey ?

Also, the deploy command won't run if the `build/web` directory isn't created.

Do we need `build:release:output`?

Reply:
Why? We use a `before_deploy` step.